### PR TITLE
docs(auth): social-login/Google - Nonce hashing requirement for successful validation

### PR DESCRIPTION
**Pull request by fabianuribe**
Documentation Update: When using a nonce, Google's ID token will contain the raw value of the provided nonce, however supabase's GoTrue expects a hash of the nonce. Providing a raw nonce to both ends will result on an  "invalid nonce"/ "Nonces mismatch" error (https://github.com/supabase/gotrue/blob/5caacc1f81ff20f8f09f4598b510c767e589a1c5/internal/api/token_oidc.go#L177)

In order for validation to work, a hashed version of the nonce should be provided to Google and a non-hashed version to the Supabase client.

## What kind of change does this PR introduce?

Docs update 

## What is the current behavior?

Current documentation does not mention the hash expectation

## What is the new behavior?

Document the expectation, provide a sample implementation for successful validation.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by gregnr** [comment]
Thanks @fabianuribe! cc @supabase/auth to confirm this change.

File: apps/docs/pages/guides/auth/social-login/auth-google.mdx [hunk 1]
```diff
@@ -289,6 +289,37 @@ async function handleSignInWithGoogle(response) {
...
+  .map((b) => b.toString(16).padStart(2, '0')) // Convert the hash to a hexadecimal string
+  .join('');
+
+// Use 'nonce' when passing to Supabase's client
+// Use 'hashedNonce' when passing to Google's authentication request
+```
+
+By following this approach, you'll ensure that the nonces are handled in a way that is compatible with both Google's and Supabase's expectations, allowing successful authentication.
+
+
```
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/social-login/auth-google.mdx:322]:
```suggestion
Supabase Auth expects the nonce in Google's ID token to be hashed (specifically with SHA-256 and represented as a hexadecimal string).

To succeed with nonce validation, you will need to provide a hashed version of the nonce to Google and the non-hashed version to the `signInWithIdToken` method.

The following code snippet provides an example of how to create both versions in a web application:

```js
// Adapted from https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#converting_a_digest_to_a_hex_string

import crypto from 'crypto';

...

const nonce = crypto.randomBytes(16).toString('base64'); // Generate a random nonce
const encoder = new TextEncoder();
const encodedNonce = encoder.encode(nonce); // Encode the nonce
const hash = await crypto.subtle.digest('SHA-256', encodedNonce); // Hash it with SHA-256
const bytes = new Uint8Array(hash);
const hashedNonce = Array.from(bytes)
  .map((b) => b.toString(16).padStart(2, '0')) // Convert the hash to a hexadecimal string
  .join('');
  
// Use 'hashedNonce' when making the authentication request to Google
// Use 'nonce' when invoking signInWithIdToken

```


```

File: apps/docs/pages/guides/auth/social-login/auth-google.mdx [hunk 2]
```diff
@@ -289,6 +289,33 @@ async function handleSignInWithGoogle(response) {
 
 Use of nonce is recommended, though optional. Make sure each nonce is generated randomly and available both in the `data-nonce` attribute as well as in the `handleSignInWithGoogle` callback function; otherwise the ID token will not be accepted.
 
+
+### Important Note on Nonce Validation
```
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/social-login/auth-google.mdx:293]:
```suggestion
### Important Note on Nonce Validation

```

File: apps/docs/pages/guides/auth/social-login/auth-google.mdx [hunk 3]
```diff
@@ -289,6 +289,34 @@ async function handleSignInWithGoogle(response) {
...
+const encoder = new TextEncoder();
+const encodedNonce = encoder.encode(nonce); // Encode the nonce
+const hash = await crypto.subtle.digest('SHA-256', encodedNonce); // Hash it with SHA-256
+const bytes = new Uint8Array(hash);
+const hashedNonce = Array.from(bytes)
+  .map((b) => b.toString(16).padStart(2, '0')) // Convert the hash to a hexadecimal string
+  .join('');
+  
+// Use 'hashedNonce' when making the authentication request to Google
+// Use 'nonce' when invoking the supabase.auth.signInWithIdToken() method
```
Comment (kangmingtay) [inline apps/docs/pages/guides/auth/social-login/auth-google.mdx:318]:
```suggestion
// Use 'nonce' when invoking the supabase.auth.signInWithIdToken() method
```
```

**Comment by kangmingtay** [comment]
@fabianuribe thanks for the prettier fix 😂

**Comment by fabianuribe** [comment]
> @fabianuribe thanks for the prettier fix 😂

Hopefully third time is the charm 🤞

**Review by kangmingtay — APPROVED** [review]

**Review by gregnr — APPROVED** [review]

**Pull request merged**